packages/sentry/index.tsx

```diff
@@ -27,3 +27,11 @@ export const sentryFetch: typeof fetch = async (resource, options) => {
          throw error;
       });
 };
+
+type PageContextProps = {
+   url: string;
+};
+
+export const setSentryPageContext = ({ url }: PageContextProps) => {
+   Sentry.setTag('resolved_url', url);
```

Comment:
I think this will be inconsistent and flaky. We're in the context of a JS server that does a bunch of async work. This JS server has only one global context yet it may be serving several requests in parallel. I think we can kill two birds though. I'm gonna try something.

Reply:
> I think this will be inconsistent and flaky. We're in the context of a JS server that does a bunch of async work. This JS server has only one global context yet it may be serving several requests in parallel.

I'm not sure I understand. What exactly are you worried about?

Feel free to push to this branch if you come up with something 👍🏻

Reply:
Wow... looks like they already auto-inject code to wrap `getServerSideProps()` (given that pull request). But it seems (based on experimentation) that they don't capture the relevant url? I wonder if there's a mechanism (via config) to augment or add to the code that auto-wraps `getServerSideProps()`.

Reply:
> But it seems (based on experimentation) that they don't capture the relevant url?

That's what I had seen, but now that I'm testing this branch, I see the correct `url` tag.

![image](https://user-images.githubusercontent.com/52104630/183747474-ed215fa7-a35a-4260-9c65-ec762f587b1a.png)


> I wonder if there's a mechanism (via config) to augment or add to the code that auto-wraps getServerSideProps().

Maybe via the middleware concept that @jarstelfox was talking about? https://github.com/iFixit/react-commerce/pull/600#discussion_r941570980

Reply:
The part that is setting the url is: https://docs.sentry.io/platforms/javascript/guides/nextjs/configuration/integrations/default/#httpcontext-previously-useragent

If you look at the code it's just grabbing it from event.request:
https://github.com/getsentry/sentry-javascript/blob/ac2dc121226e0b1aa1fc199e5b7549862fc99f5f/packages/browser/src/integrations/httpcontext.ts#L31

which is set here: https://github.com/getsentry/sentry-javascript/blob/ac2dc121226e0b1aa1fc199e5b7549862fc99f5f/packages/nextjs/src/utils/withSentry.ts#L44

tl;dr: I *think* it should just work, I don't know why we are having issues.

Reply:
Great thanks for finding all that! [Here's an example](https://sentry.io/share/issue/7b644c341c5f4359a5712f3025733c2b/) of a Sentry event from `getServerSideProps` that didn't get the `url` tag. But many other events from the server are. I can't explain what's causing the discrepancy, I agree with you that it seems like it should work.

Reply:
I see that it's setting the url, but it isn't setting resolvedUrl. Is there something in this code that makes you think we should be seeing resolvedUrl already?

Reply:
Okay, I did some skimming and cross-referenced the example you linked above.

It looks like sentry hooks into the server and wraps a ton of logic. Most notably [Next's Request Handler Factory](https://github.com/getsentry/sentry-javascript/blob/ac2dc121226e0b1aa1fc199e5b7549862fc99f5f/packages/nextjs/src/utils/instrumentServer.ts#L147-L149) which is what [calls down to setting the url](https://github.com/getsentry/sentry-javascript/blob/ac2dc121226e0b1aa1fc199e5b7549862fc99f5f/packages/nextjs/src/utils/instrumentServer.ts#L247).

I still believe it should be working at the time of the error(even though it clearly is not).

Sentry is passing that data down to itself when a request handler is made. i.e. `afterRouted`

Maybe someone else can spy what's missing, but I think we should provide our own way of getting the data until we can find a reliable way for sentry to hook the url

Reply:
Just to make sure we're all on the same page, this should only affect page loads where it uses next-link to load json. In those cases, we've seen url set as `/_next/...`. `resolvedUrl` will be the url of the page that we were really trying to link to. In the code I've read, I see sentry adding the url field, but I haven't seen them add the resolvedUrl field, so I wouldn't expect it to just work without some changes like in this pull.

Reply:
Hi, Sentry JS SDK maintainer here. Just stumbled onto this thread.

We're currently working on wrapping Next.js data fetching methods (including `getServerSideProps`) to improve instrumentation. Currently, the efforts mainly go into properly setting parameterized transaction names (transactions are part of the Sentry performance monitoring product and parameterization means that transactions won't show up as `/user/42/post/1337` but rather as `/user/[userId]/post/[postId]`, which is generally more useful).

We might explore adding the parameterized route + parameters to errors in some way (or at least provide a way for SDK user's to figure out what parameterized route they're in - maybe we can contribute this to the Next.js repo upstream).

We would love for you to try out the changes as soon as they land!